model/src/commonMain/kotlin/JobConfigurations.kt

```diff
@@ -80,7 +80,7 @@ data class AnalyzerJobConfiguration(
     /**
      * The list of package curation providers to use.
      */
-    val packageCurationProviders: List<ProviderPluginConfiguration> = emptyList(),
+    val packageCurationProviders: List<ProviderPluginConfiguration>? = null,
```

Comment:
I find it strange that we need to add quirks to our data model, just to make the Swagger generation work. If this is really required, should this change then be limited to the API module? There is no need for the property in `model` to be nullable, too.

Reply:
I thought about that too, but I wanted to be consistent, and so far all properties have the same types and default values in both models. We could change it for lists, but then we should change it for all of them.

Reply:
It was the original purpose of this mapping to separate the internal model from the API. IMHO, there can well be differences if this helps us. Avoiding unnecessary null checks would fall into this category.

Reply:
I took a look into making more properties non-nullable in the model module, but that lead me to lot of questions how to handle some inconsistencies we already have in the job configurations. I don't want to go down this rabbit hole right now, so I'll keep the model properties non-nullable in this PR, even if it makes the current state of the mapping even more inconsistent. At least I could make some cleanups during the investigation for which I will make a separate PR after this is merged.

Reply:
@oheger-bosch Done.